### PR TITLE
Fixes for Aromatherapy, Heal Bell, Doom Desire and Future Sight

### PR DESCRIPTION
**Pull request by Slayer95**

### CODE REVIEW

**Comment by Joimer** [comment]
:^)

**Comment by Zarel** [comment]
Is this still relevant after other recent refactors?

File: battle-engine.js [hunk 1]
```diff
@@ -2204,20 +2204,23 @@ Battle = (function () {
...
-		if (status[callbackType] !== undefined || (getAll && thing.itemData[getAll])) {
-			statuses.push({status: status, callback: status[callbackType], statusData: thing.itemData, end: thing.clearItem, thing: thing});
-			this.resolveLastPriority(statuses, callbackType);
-		}
-		status = this.getEffect(thing.template.baseSpecies);
-		if (status[callbackType] !== undefined) {
-			statuses.push({status: status, callback: status[callbackType], statusData: thing.speciesData, end: function () {}, thing: thing});
-			this.resolveLastPriority(statuses, callbackType);
+		if (thing.isActive || this.gen <= 4) {
+			// inactive effects for inactive Pokémon since Gen 5
```
Comment (Zarel) [inline battle-engine.js]: I disagree that inactive effects should be run in the event handler. I'd rather hardcode them in the rare situations they come up.

Comment (Slayer95) [inline battle-engine.js]: Fair enough. That probably implies that this should be just
<code> if (thing.isActive) ...</code>.

The hardcoded part would be removing Soundproof pokémon from the list of sound-based targets for <code>allyTeam</code>.

**Comment by Slayer95** [comment]
Yes, it's still relevant

File: battle-engine.js (hunk 1 above)
Comment (Zarel) [inline battle-engine.js]: In theory `allyTeam` should just target the user and make the move's `Hit` handler do the work of iterating through the team. How many `allyTeam` moves do we actually have?

Comment (Zarel) [inline battle-engine.js]: Or at least that's how I'd implement it. That's how `allySide` moves are implemented, iirc.

**Comment by Slayer95** [comment]
Not anymore: https://github.com/Zarel/Pokemon-Showdown/commit/d58ddc671be21049d53903caa0c91e916b1b140c

The fixes for future moves only apply under certain circumstances. The proper way to fix them is to precalculate the damage on move usage.

**Pull request closed without merging**